gpiozero/pins/data.py

```diff
@@ -422,13 +782,185 @@ class PiBoardInfo(namedtuple('PiBoardInfo', (
+            # RRRR     - Revision (0, 1, 2, etc.)
+            try:
+                model = {
+                    0: 'A',
+                    1: 'B',
+                    2: 'A+',
+                    3: 'B+',
+                    4: '2B',
+                    6: 'CM',
+                    8: '3B',
+                    9: 'Zero',
+                    }[(revision & 0xff0) >> 4]
+                if model in ('A', 'B'):
+                    pcb_revision = {
+                        0: '1.0', # is this right?
+                        1: '1.0',
+                        2: '2.0',
+                        }[revision & 0x0f]
+                else:
+                    pcb_revision = '1.%d' % (revision & 0x0f)
+                released = {
+                    'A':    '2013Q1',
+                    'B':    '2012Q1' if pcb_revision == '1.0' else '2012Q4',
+                    'A+':   '2014Q4',
+                    'B+':   '2014Q3',
+                    '2B':   '2015Q1',
+                    'CM':   '2014Q2',
+                    '3B':   '2016Q1',
+                    'Zero': '2015Q4' if pcb_revision == '1.2' else '2016Q2',
+                    }[model]
+                soc = {
+                    0: 'BCM2835',
+                    1: 'BCM2836',
+                    2: 'BCM2837',
+                    }[(revision & 0xf000) >> 12]
+                manufacturer = {
+                    0: 'Sony',
+                    1: 'Egoman',
+                    2: 'Embest',
+                    }[(revision & 0xf0000) >> 16]
+                memory = {
+                    0: 256,
+                    1: 512,
+                    2: 1024,
+                    }[(revision & 0x700000) >> 20]
+                storage = {
+                    'A': 'SD',
+                    'B': 'SD',
+                    'CM': 'eMMC',
+                    }.get(model, 'MicroSD')
+                usb = {
+                    'A':    1,
+                    'A+':   1,
+                    'Zero': 1,
+                    'B':    2,
+                    'CM':   1,
+                    }.get(model, 4)
+                ethernet = {
+                    'A':    0,
+                    'A+':   0,
+                    'Zero': 0,
+                    'CM':   0,
+                    }.get(model, 1)
+                wifi = {
+                    '3B': True,
+                    }.get(model, False)
+                bluetooth = {
+                    '3B': True,
+                    }.get(model, False)
+                csi = {
+                    'Zero': 0 if pcb_revision == '1.2' else 1,
+                    'CM':   2,
+                    }.get(model, 1)
+                dsi = {
+                    'Zero': 0,
+                    }.get(model, csi)
+                headers = {
+                    'A':  {'P1': REV2_P1, 'P5': REV2_P5},
+                    'B':  {'P1': REV1_P1} if pcb_revision == '1.0' else {'P1': REV1_P1, 'P5': REV2_P5},
```

Comment:
The bit at the very end of this line should read `{'P1': REV2_P1, 'P5': REV2_P5},`  (you've got REV1_P1 instead of REV2_P1)